app/jobs/obtain_annotations_with_callback_job.rb

```diff
@@ -0,0 +1,125 @@
+class ObtainAnnotationsWithCallbackJob < ApplicationJob
+  include UseJobRecordConcern
+  include ActionView::Helpers::NumberHelper
+
+  queue_as :low_priority
+
+  def perform(project, filepath, annotator, options)
+    line_count = File.read(filepath).each_line.count
+
+    prepare_progress_record(line_count)
+
+    # for asynchronous protocol
+    single_doc_processing_p = annotator.single_doc_processing?
+
+    doc_collection = DocCollection.new
+    File.foreach(filepath) do |line|
+      docid = line.chomp.strip
+      doc = Doc.find(docid)
+      doc.set_ascii_body if options[:encoding] == 'ascii'
+
+      if doc_collection.present? && (single_doc_processing_p || (doc_collection.size + doc.body.length) > annotator.find_or_define_max_text_size)
+        begin
+          handle_annotate_request(annotator, project, doc_collection.docs, options)
+        rescue Exceptions::JobSuspendError
+          raise
+        rescue => e
+          less_docs_message = 'Could not obtain annotations:'
+          many_docs_message = 'Could not obtain annotations for'
+          add_exception_message_to_job(doc_collection.docs, e, less_docs_message, many_docs_message)
+        ensure
+          doc_collection.clear
+        end
+      end
+
+      doc_collection << doc
+
+    rescue Exceptions::JobSuspendError
+      raise
+    rescue RuntimeError => e
+      less_docs_message = 'Runtime error:'
+      many_docs_message = 'Runtime error while processing'
+      add_exception_message_to_job(doc_collection.docs, e, less_docs_message, many_docs_message)
+    end
+
+    if doc_collection.present?
```

Comment:
```suggestion
    if doc_collection.rest?
```
が良いです。
残りがあったら実行するイメージです。